packages/upload/src/FzUpload.vue

```diff
@@ -68,12 +68,33 @@ const props = withDefaults(defineProps<FzUploadProps>(), {
   buttonLabel: "Carica",
   dragAndDropLabel: "o trascina qui",
 });
-const emit = defineEmits([]);
-const model = defineModel<File[]>();
+const emit = defineEmits(["fzupload:change"]);
+const model = defineModel<File[]>({
+  default: [],
+});
 
 const input = ref<HTMLInputElement | null>(null);
 const urlByFileMap = ref(new Map<File, string>());
 
+onMounted(() => {
+  if (model.value && model.value.length > 0 && !props.multiple) {
+    console.warn(
+      "[Fiscozen Design System]: FzUpload prop 'multiple' is set to false, but multiple files are provided. Only the first file will be used.",
+    );
+    model.value = [model.value[0]];
+  }
+});
+
+watch(
+  () => model.value,
+  (newFiles: File[]) => {
+    if (newFiles.length > 1 && !props.multiple) {
+      model.value = [newFiles[0]];
+    }
+    emit("fzupload:change", model.value);
```

Comment:
I think this is redundant, there's already update:modelValue which does the exact same thing

Reply:
emitting the change allows the component to be used without passing a model. It is also easier to understand than ‘update:modelValue’.